### PR TITLE
Usability

### PR DESCRIPTION
**Pull request by nikhilwoodruff**
A few things for ease of use:

- Count function (needs testing, probably)
- Overrides for common magic methods: microseriesA + microseriesB = microseriesC with weights from microseries A, (before: = pd.Series)
- Weight*s*_col
- Potential function decorator for parsing column names into microseries in the dataframe: @MaxGhenis I added a simple example, poverty headcount - what do you think?

### CODE REVIEW

**Comment by MaxGhenis** [comment]
Too cool. I didn't know Python could do this. Thanks @nikhilwoodruff. 

Could you just add a docstring and type hints for `poverty_headcount`? Think it requires `typing.Union`.

File: microdf/generic.py [hunk 1]
```diff
@@ -225,3 +322,24 @@ def groupby(self, by, *args, **kwargs):
...
+
+        :param threshold: threshold array or column name
+        :type threshold: Union[MicroSeries, str]
+
+        return: number of entities in poverty
+        rtype: int
+        """
+        in_poverty = income < threshold
+        x = in_poverty.sum()
+        return x.count()
```
Comment (MaxGhenis) [inline microdf/generic.py:345]: It's not just `x`? Could you add a test?

**Review by MaxGhenis — COMMENTED** [review]
.

File: microdf/generic.py (hunk 1 above)
Comment (nikhilwoodruff) [reply microdf/generic.py:345]: Yep, you're right, it should be just x here

**Comment by nikhilwoodruff** [comment]
Should now fix #161 and #167

**Comment by MaxGhenis** [comment]
Some issues with `groupby`:

`groupby(list)` breaks:
```
mdf.MicroDataFrame({"x": [1, 2], "y": [3, 4], "z": [5, 6]}).groupby(["x", "y"]).z.sum()
```
>KeyError: "None of [Index(['x', 'y'], dtype='object')] are in the [index]"

and with a single groupby, the result is wrong:
```
mdf.MicroDataFrame({"x": [1, 2], "y": [3, 4], "z": [5, 6]}).groupby("x").z.sum()
```
```
x
1    0.0
2    0.0
dtype: float64
```
This works in pandas:
```
pd.DataFrame({"x": [1, 2], "y": [3, 4], "z": [5, 6]}).groupby(["x", "y"]).z.sum()
```
```
x  y
1  3    5
2  4    6
Name: z, dtype: int64
```

**Comment by MaxGhenis** [comment]
Fixes #173 and #174 right?

**Pull request merged**